src/current/v23.1/import-performance-best-practices.md

```diff
@@ -89,16 +52,72 @@ This method has the added benefit of alerting on potential issues with the impor
+## Split your data into multiple files
+
+Splitting the import data into multiple files can have a significant impact on the import performance. The following formats support multi-file import using `IMPORT INTO`:
+
+- `CSV`
+- `DELIMITED DATA`
+- `AVRO`
+
+For these formats, we recommend splitting your data into at least as many files as there are nodes.
+
+For example, if you have a 3-node cluster, split your data into at least 3 files, create your table schema, and [import into that table](import-into.html):
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE TABLE customers (id UUID PRIMARY KEY, name TEXT, INDEX name_idx(name));
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+IMPORT INTO customers (id, name)
+    CSV DATA (
+      's3://{BUCKET NAME}/{customers.csv}?AWS_ACCESS_KEY_ID={KEY ID}&AWS_SECRET_ACCESS_KEY={SECRET ACCESS KEY}',
+      's3://{BUCKET NAME}/{customers2.csv}?AWS_ACCESS_KEY_ID={KEY ID}&AWS_SECRET_ACCESS_KEY={SECRET ACCESS KEY}',
+      's3://{BUCKET NAME}/{customers3.csv}?AWS_ACCESS_KEY_ID={KEY ID}&AWS_SECRET_ACCESS_KEY={SECRET ACCESS KEY}',
+      's3://{BUCKET NAME}/{customers4.csv}?AWS_ACCESS_KEY_ID={KEY ID}&AWS_SECRET_ACCESS_KEY={SECRET ACCESS KEY}',
+    );
+~~~
+
+CockroachDB imports the files that you give it, and does not further split them. For example, if you import one large file for all of your data, CockroachDB will process that file on one node, even if you have more nodes available. However, if you import three files (and your cluster has at least three nodes), each node will process a file in parallel. This is why splitting your data into at least as many files as you have nodes will dramatically decrease the time it takes to import data.
+
+{{site.data.alerts.callout_info}}
+You can split the data into **more** files than you have nodes. CockroachDB will process the files in parallel across the cluster. When splitting the data Cockroach Labs recommends splitting it into a multiple of the number of nodes in your cluster, if possible. For example, if you have a 3 node cluster, split the dataset into 9, 27, or 300 files.
+
+Cockroach Labs recommends keeping the files to a maximum file size of 4 GB, and to keep each file size similar across the dataset. For example, if you are importing a 9 GB dataset that was split into 3 files into a 3 node cluster, keep each file around 3 GB in size if possible. Don't split the data into two 4 GB files, and one 1 GB file.
+{{site.data.alerts.end}}
+
+For maximum performance each split file should be *partitioned*, meaning that if you were to sort the rows in each file there would be no overlapping data in any other file. For example, if your dataset had an alphabetic string primary key and you were importing the data into a 3 node cluster, you would split the files so that the first file contained rows where the primary key began with the letter "a" to "i", the second file "j" to "r" and the third file "s" to "z". No file should contain duplicate rows.
```

Comment:
I think a visual example might be useful here

Reply:
I used the term "partitioned" when I was doing IMPORT performance work. I think it might have been confusing for folks, even internally. Another way to express this might be "sorted within a file" and "sorted across all files".